cli/cmd/install.go

```diff
@@ -207,6 +208,7 @@ func newCmdInstall() *cobra.Command {
 	// the configuration in validateAndBuild.
 	flags := options.recordableFlagSet()
 	installOnlyFlags := options.installOnlyFlagSet()
+	disableIdentityFlag := options.proxyConfigOptions.disableIdentityFlagSet()
```

Comment:
I'd think we want this flag to be recorded. Otherwise, during upgrade, [identity will be re-introduced](https://github.com/linkerd/linkerd2/blob/85a7c885d47f17635b8cc53ee97f724fd5dedbeb/cli/cmd/upgrade.go#L175-L176). Recording this flag will allow us to distinguish what `if idctx == nil` mean during upgrade, i.e. upgrading from a version that didn't support identity vs. identity is disabled.

Reply:
Looks like this flag is already recorded:
```
   install: |
     {"uuid":"8dcc640d-76b4-4ef9-953b-977a042b15b8","cliVersion":"git-f12d824c","flags":[{"name":"disable-identity","value":"true"}]}
```

I wonder if we should add it to the recordable flagset? Then it will also be available for upgrade. That way, a user will be able to enable identity during upgrade.

Reply:
I haven't really explored the option of enabling identity during an upgrade. All the `identity-issuer-*` and `identity-trust-*` flags are disabled in `linkerd upgrade`, I guess because it's presumed that was already set during `linkerd install`. This also entails the question whether `--disable-identity=true` issued during an upgrade would cause problems. If so we could only allow enabling identity during upgrades and not disabling. But that's something I'd rather do in a separate PR.

Reply:
> All the identity-issuer-* and identity-trust-* flags are disabled in linkerd upgrade

Right, because during upgrade, those identity assets shouldn't change. Otherwise, the existing data plane will be broken. Upgrade will either generate new ones (if none exist), or re-use existing ones.